enumerables.rb

```diff
@@ -0,0 +1,67 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
```

Comment:
- returns an array containing all elements of enum for which the given block returns a true value, kindly fix it to make it work with a range.
TIP: you can convert it into an array